clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/VeniceAdaptiveIngestionThrottler.java

```diff
@@ -71,13 +72,15 @@ public void checkSignalAndAdjustThrottler() {
         hasLimitedRate = true;
         isSignalIdle = false;
         signalIdleCount = 0;
-        if (currentThrottlerIndex > 0) {
-          currentThrottlerIndex--;
+        int currentIndex = currentThrottlerIndex.get();
+        // for limiter signal, step down limit in faster rate.
+        if (currentIndex >= 1) {
+          currentThrottlerIndex.set(Math.max(currentIndex - 2, 0));
```

Comment:
Is it possible that we want even faster than 2x? But I think it is OK to hardcode as 2 for testing now.

Reply:
We can, let me try the decrement by 2 first.